pkg/services/ngalert/schedule/schedule.go

```diff
@@ -368,7 +368,11 @@ func (sch *schedule) schedulePeriodic(ctx context.Context) error {
 	for {
 		select {
 		case tick := <-sch.ticker.C:
-			start := time.Now()
+			// We use Round(0) on the start time to remove the monotonic clock.
+			// This is required as late ticks from the ticker have current monotonic
+			// timestamps such that start.Sub(tick) does not return the expected
+			// delta.
```

Comment:
```suggestion
			// This is required as late ticks from the ticker contain a monotonic clock reading
			// and so does `start`, therefore operations such as `start.Sub(tick)` will use the monotonic values
			// and ignore wall time.
```

Reply:
The above is my understanding from:

```
The Time returned by time.Now contains a monotonic clock reading. If Time t has a monotonic clock reading, t.Add adds the same duration to both the wall clock and monotonic clock readings to compute the result. Because t.AddDate(y, m, d), t.Round(d), and t.Truncate(d) are wall time computations, they always strip any monotonic clock reading from their results. Because t.In, t.Local, and t.UTC are used for their effect on the interpretation of the wall time, they also strip any monotonic clock reading from their results. The canonical way to strip a monotonic clock reading is to use t = t.Round(0).

If Times t and u both contain monotonic clock readings, the operations t.After(u), t.Before(u), t.Equal(u), and t.Sub(u) are carried out using the monotonic clock readings alone, ignoring the wall clock readings. If either t or u contains no monotonic clock reading, these operations fall back to using the wall clock readings.

On some systems the monotonic clock will stop if the computer goes to sleep. On such a system, t.Sub(u) may not accurately reflect the actual time that passed between t and u.
```

TIL more about monotonic clocks than what I wanted to know.